alf/environments/parallel_environment.py

```diff
@@ -157,6 +157,9 @@ def reward_spec(self):
     def time_step_spec(self):
         return self._time_step_spec
 
+    def render(self, mode="rgb_array"):
```

Comment:
Why do we need to do parallel  environment render?

Reply:
This is because Wei is correct.  We cannot render to rgb using mujoco egl and then start a pygame display at the same time, unless the environment is in a separate process.
alf.environment.create_environment creates parallelenv wrapped processenv.  That's why ParaEnv needs to pass render along from the child process.

Reply:
Perhaps you can directly use ProcessEnvironment instead of ParallelAlfEnvironment?

Reply:
I tried using ProcessEnvironment directly.

There is a lot of logic inside ParallelEnvironment that needs to be copied: batching, tensor convert from np to torch and from cpu to GPU, (and also custom logic to handle accessing class members like "wrapped_env" and "console_info" etc..)  Uglier than the current hacking of ParallelEnv.

I could separate such duplicate logic into a ProcessAdaptorEnvironment, so ParallelEnv will be more dedicated to proc start & stop, and ProcAdaptorEnv can be easily used standalone (wrapping outside ProcessEnvironment).  But that incurs additional compute, several nest loops.  (There probably still needs to be some custom logic handling "render" and "metadata" etc..)

Let me know which you prefer, or any better idea you may have.  @emailweixu